mod/tos.php

```diff
@@ -0,0 +1,61 @@
+<?php
+/**
+ * @file mod/tos.php
+ *
+ * This module displays the Terms of Service for a node, if the admin
+ * wants them to be displayed.
+ */
+use Friendica\App;
+use Friendica\Core\Config;
+use Friendica\Core\L10n;
+use Friendica\Core\System;
+use Friendica\Module\Login;
+use Friendica\Content\Text\BBCode;
+
+/**
+ * @brief initialize the TOS module.
+ *
+ * If this is a single user instance, we expect the user to know their
+ * dealings with their own node so a TOS is not necessary.
+ *
+ * @param App $a
+ * */
+function tos_init(App $a) {
+
+	if (strlen(Config::get('system','singleuser'))) {
+		goaway(System::baseUrl()."/profile/" . Config::get('system','singleuser'));
+	}
+
+}
+
+/**
+ * @brief generate the content of the /tos page
+ *
+ * The content of the /tos page is generated from two parts.
+ * (1) a free form part the admin of the node can set in the admin panel
+ * (2) an optional privacy statement that gives some transparency about
+ *     what information are needed by the software to provide the service.
+ *     This privacy statement has fixed text, so it can be translated easily.
+ *
+ * @param App $a
+ * @return string
+ */
+function tos_content(App $a) {
```

Comment:
Can you please move this new module to the `src/Module/` folder by taking example on the new classes already there? The `mod/` folder is destined to disappear altogether as the `include/` folder.

Reply:
No. I studied the files and I tried to put the module into src/Module but it does not work. Is there any properly documented example for dummies?

Reply:
The basics: for `/tos` query string, you want a `src/Module/Tos.php` file with a `class Tos extends Module` inside.

Then copy the content of `tos_content()` in `Tos::content()`, remember to add a `$a = self::getApp()` if you are using `$a`.

In the same fashion, copy the content of `tos_init()` in `Tos::init()` with the same caveat.

Reply:
Yes, that was the obvious part. The naming was the hidden secret. I added a hint about the naming to the BaseModule for future reference.

Reply:
Thanks!